lib/elixir/test/elixir/macro_test.exs

```diff
@@ -620,6 +621,57 @@ defmodule MacroTest do
     assert Macro.unpipe(quote(do: foo |> bar |> baz)) == quote(do: [{foo, 0}, {bar, 0}, {baz, 0}])
   end
 
+  # guard
+
+  test "guard" do
```

Comment:
It would be nice to test the error messages too and break this test into multiple ones (describe can help!).